s3-access-grants-plugin/src/main/java/com/amazonaws/s3accessgrants/plugin/S3AccessGrantsRequestHandler.java

```diff
@@ -51,6 +52,7 @@ public class S3AccessGrantsRequestHandler {
     private static final Log logger = LogFactory.getLog(S3AccessGrantsRequestHandler.class);
     private final boolean enableCrossRegionAccess;
     private ConcurrentHashMap<Regions, AWSS3Control> clientsCache = new ConcurrentHashMap<>();
+    private ClientConfiguration clientConfiguration = new ClientConfiguration().withUserAgentPrefix("aws-s3-accessgrants-java-sdk-v1-plugin");
```

Comment:
what happens if another user agent is used? no cache? unlimited cache? or does it only allow the `aws-s3-accessgrants-java-sdk-v1-plugin` ?

Reply:
We are setting the user agent here. So any API calls made with this client will have the user agent as "aws-s3-accessgrants-java-sdk-v1-plugin".